pepper-apis/src/main/resources/db-changes/schema/DDP-7069-matrix-question-type.xml

```diff
@@ -0,0 +1,179 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<databaseChangeLog xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+                   http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.5.xsd">
+
+    <changeSet author="ypavlov" id="matrix-select-mode-table">
+        <createTable tableName="matrix_select_mode">
+            <column name="matrix_select_mode_id" type="bigint" autoIncrement="true" startWith="1">
+                <constraints primaryKey="true" primaryKeyName="matrix_select_mode_pk"/>
+            </column>
+            <column name="matrix_select_mode_code" type="varchar(45)">
+                <constraints nullable="false" unique="true" uniqueConstraintName="matrix_select_mode_code_uk"/>
+            </column>
+        </createTable>
+    </changeSet>
+
+    <changeSet author="ypavlov" id="matrix-question-table">
+        <createTable tableName="matrix_question">
+            <column name="question_id" type="bigint">
+                <constraints primaryKey="true" primaryKeyName="matrix_question_pk"
+                             references="question(question_id)" foreignKeyName="matrix_question_question_fk"/>
+            </column>
+            <column name="matrix_select_mode_id" type="bigint">
```

Comment:
Wondering if select mode : SINGLE is exactly same as option: IS_EXCLUSIVE if the select mode scope is entire question and NOT option in a ROW !!

Reply:
Just like in PICKLIST question, we can use SINGLE (radio buttons) and MULTIPLE (checkboxes) with IS_EXCLUSIVE. However, there is a difference other than look & feel: radio button cannot be cleared.